protocol/contracts/nexus/BaseSacrifice.sol

```diff
@@ -0,0 +1,109 @@
+pragma solidity 0.8.19;
+// SPDX-License-Identifier: AGPL-3.0-or-later
+// Temple (nexus/BaseSacrifice.sol)
+
+
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { IBaseSacrifice } from "../interfaces/nexus/IBaseSacrifice.sol";
+import { ElevatedAccess } from "./access/ElevatedAccess.sol";
+import { CommonEventsAndErrors } from "../common/CommonEventsAndErrors.sol";
+import { mulDiv } from "@prb/math/src/Common.sol";
+
+abstract contract BaseSacrifice is IBaseSacrifice, ElevatedAccess {
+    /// @notice start time from which price increases
+    uint64 public originTime;
+    /// @notice custom price set by governance
+    uint256 public customPrice;
+
+    uint256 private constant MINIMUM_CUSTOM_PRICE = 30 ether;
+    uint256 private constant ONE_ETHER = 1 ether;
+
+    /// @notice Price parameters 
+    PriceParam public priceParams;
+
+    /*
+     * @notice Set origin time.
+     * Origin time is the start of the linear ascending price to params.priceMaxPeriod
+     * @param _originTime Origin time
+     */
+    function setOriginTime(uint64 _originTime) external override onlyElevatedAccess {
+        if (_originTime < block.timestamp) { revert CommonEventsAndErrors.InvalidParam(); }
+        originTime = _originTime;
+        emit OriginTimeSet(originTime);
+    }
+
+    /*
+     * @notice Set price parameters.
+     * @param _priceParams Price parameters to set
+     */
+    function setPriceParams(PriceParam calldata _priceParams) external override onlyElevatedAccess {
+        if (_priceParams.minimumPrice > _priceParams.maximumPrice) { revert CommonEventsAndErrors.InvalidParam(); }
+        if (_priceParams.minimumPrice < MINIMUM_CUSTOM_PRICE) { revert CommonEventsAndErrors.InvalidParam(); }
+        if (_priceParams.priceMaxPeriod == 0) { revert CommonEventsAndErrors.InvalidParam(); }
+        priceParams.priceMaxPeriod = _priceParams.priceMaxPeriod;
+        priceParams.minimumPrice = _priceParams.minimumPrice;
+        priceParams.maximumPrice = _priceParams.maximumPrice;
+
+        emit PriceParamsSet(_priceParams);
+    }
+
+    /*
+     * @notice Set custom price
+     * owner can reset price with 0 _price value. Custom price can be set anytime during or after params.priceMaxPeriod on
+     * a flash sale or at a discounted price.
+     * @param _price Custom price 
+     */
+    function setCustomPrice(uint256 _price) external virtual onlyElevatedAccess {
+        if (_price != 0 && _price < MINIMUM_CUSTOM_PRICE) { revert CommonEventsAndErrors.InvalidParam(); }
+        customPrice = _price;
+        emit CustomPriceSet(customPrice);
+    }
+
+    /*
+     * @notice Sacrifice tokens to mint a Relic
+     * Caller must approve contract to spend tokens
+     * @param enclaveId Enclave ID
+     */
+    function sacrifice(uint256 enclaveId) external virtual {}
```

Comment:
Do you need both? Could always just use the one which takes the recipient?

Then if it's in the interface, you can leave them out of the BaseSacrifice.sol contract, which means the derived contract will be forced (at compile time) to implement them rather than accidentally miss it